### PR TITLE
Meta: Attach built extension to releases

### LINKED ISSUE #4595: Include artifact in release workflows
In the `Version` job, we should also use `upload-artifact` like we already do in the Test workflow

### PR DESCRIPTION
**Pull request by mowies**
### This PR
- attaches the `distribution` folder created by `npm test` to the created release
- archives the `distribution` folder into a tarball before attaching to the release

Closes #4595

### CODE REVIEW

**Comment by yakov116** [comment]
Not by a computer to test. Are you attaching the refined-github.zip file to the release?

**Comment by mowies** [comment]
@yakov116  Nope, I'm just attaching whatever gets generated by `npm test` into the `distribution/` folder as a `tar.gz` archive.
That's what was asked for in the issue here: #4595

**Comment by mowies** [comment]
Could you please be more specific? I didn't find a zip archive that's generated somewhere already. I suppose you mean that the bundle files in `distribution` get zipped into `refined-github.zip`?

**Comment by fregante** [comment]
I think this is right already, what's the issue?

**Comment by yakov116** [comment]
> I think this is right already, what's the issue? The filename?

I think so

**Comment by mowies** [comment]
Sure, I can change it to a zip with your mentioned name @yakov116

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -30,14 +30,12 @@ jobs:
...
-          github-token: ${{ secrets.GITHUB_TOKEN }}
-          script: |
-            await github.request(`POST /repos/${{ github.repository }}/releases`, {
-              tag_name: "${{ steps.daily-version.outputs.version }}",
-              generate_release_notes: true
-            });
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          tar -czvf "distribution.tar.gz" "./distribution/"
```
Comment (fregante) [inline .github/workflows/release.yml:36]: It'd be good to ensure that the generated zip does not contain the folder `distribution`. Probably 

```suggestion
          cd distribution
          zip -r refined-github.zip *
```

Comment (mowies) [reply .github/workflows/release.yml]: done

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -33,8 +33,9 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
-          tar -czvf "distribution.tar.gz" "./distribution/"
-          gh release create "${{ steps.daily-version.outputs.version }}" --generate-notes "distribution.tar.gz"
+          cd distribution
+          zip -r "refined-github.zip" *
```
Comment (fregante) [inline .github/workflows/release.yml:37]: Not necessary 

```suggestion
          zip -r refined-github.zip *
```

**Review by fregante — APPROVED** [review]

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -30,14 +30,13 @@ jobs:
...
-          script: |
-            await github.request(`POST /repos/${{ github.repository }}/releases`, {
-              tag_name: "${{ steps.daily-version.outputs.version }}",
-              generate_release_notes: true
-            });
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          cd distribution
+          zip -r refined-github.zip *
```
Comment (fregante) [inline .github/workflows/release.yml:37]: @yakov116 I wonder if we should rename it to clarify that this isn't the official way to install it. Like "refined-github-local-install.zip"

We should probably also append the official download links to the releases as well, like:

> Download: [16px chrome icon], [16px firefox icon], [16px safari icon]

**Comment by fregante** [comment]
Thank you @mowies! Looking forward to more PRs like this

**Comment by fregante** [comment]
It worked (after a quick fix) but manifest.json had the version set to 0.0.0 😕 I removed the attachment:

[refined-github-local-install.zip](https://github.com/refined-github/refined-github/files/9867547/refined-github-local-install.zip)

**Comment by fregante** [comment]
I opened https://github.com/refined-github/refined-github/pull/6114 but I don't have time to test it right now. Feel free to try and open a new PR based on it if you'd like

**Pull request merged**